If you open `integration_run_0_flaky.log` in `details` of `Integration tests flaky check (asan) — Invalid check_status.tsv `
you will see:
```
=============================== warnings summary ===============================
test_materialized_mysql_database/test.py::test_mysql_killed_while_insert_5_7
  /usr/local/lib/python3.10/dist-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-1 (insert)
  
  Traceback (most recent call last):
    File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
      self.run()
    File "/usr/lib/python3.10/threading.py", line 953, in run
      self._target(*self._args, **self._kwargs)
    File "/ClickHouse/tests/integration/test_materialized_mysql_database/materialized_with_ddl.py", line 1401, in insert
      mysql_node.query(query)
    File "/ClickHouse/tests/integration/test_materialized_mysql_database/test.py", line 101, in query
      cursor.execute(execution_query)
    File "/usr/local/lib/python3.10/dist-packages/pymysql/cursors.py", line 153, in execute
      result = self._query(query)
    File "/usr/local/lib/python3.10/dist-packages/pymysql/cursors.py", line 322, in _query
      conn.query(q)
    File "/usr/local/lib/python3.10/dist-packages/pymysql/connections.py", line 558, in query
      self._affected_rows = self._read_query_result(unbuffered=unbuffered)
    File "/usr/local/lib/python3.10/dist-packages/pymysql/connections.py", line 822, in _read_query_result
      result.read()
    File "/usr/local/lib/python3.10/dist-packages/pymysql/connections.py", line 1200, in read
      first_packet = self.connection._read_packet()
    File "/usr/local/lib/python3.10/dist-packages/pymysql/connections.py", line 772, in _read_packet
      packet.raise_for_error()
    File "/usr/local/lib/python3.10/dist-packages/pymysql/protocol.py", line 221, in raise_for_error
      err.raise_mysql_exception(self._data)
    File "/usr/local/lib/python3.10/dist-packages/pymysql/err.py", line 143, in raise_mysql_exception
      raise errorclass(errno, errval)
  pymysql.err.OperationalError: (1053, 'Server shutdown in progress')
  
    warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))

test_materialized_mysql_database/test.py::test_mysql_killed_while_insert_8_0
  /usr/local/lib/python3.10/dist-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-2 (insert)
  
  Traceback (most recent call last):
    File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
      self.run()
    File "/usr/lib/python3.10/threading.py", line 953, in run
      self._target(*self._args, **self._kwargs)
    File "/ClickHouse/tests/integration/test_materialized_mysql_database/materialized_with_ddl.py", line 1401, in insert
      mysql_node.query(query)
    File "/ClickHouse/tests/integration/test_materialized_mysql_database/test.py", line 100, in query
      with self.alloc_connection().cursor() as cursor:
    File "/ClickHouse/tests/integration/test_materialized_mysql_database/test.py", line 97, in alloc_connection
      raise Exception("Connection not establised, {}".format(errors))
  Exception: Connection not establised, ['(2003, "Can\'t connect to MySQL server on \'172.16.1.3\' ([Errno 111] Connection refused)")', '(2003, "Can\'t connect to MySQL server on \'172.16.1.3\' ([Errno 111] Connection refused)")', '(2003, "Can\'t connect to MySQL server on \'172.16.1.3\' ([Errno 111] Connection refused)")', '(2003, "Can\'t connect to MySQL server on \'172.16.1.3\' ([Errno 111] Connection refused)")', '(2003, "Can\'t connect to MySQL server on \'172.16.1.3\' ([Errno 111] Connection refused)")']
```

--- a/src/Core/MySQL/MySQLReplication.cpp
+++ b/src/Core/MySQL/MySQLReplication.cpp
@@ -121,6 +121,17 @@ namespace MySQLReplication
         {
             typ = QUERY_SAVEPOINT;
         }
+
+        // https://dev.mysql.com/worklog/task/?id=13355
+        // When doing query "CREATE TABLE xx AS SELECT", the binlog will be
+        // "CREATE TABLE ... START TRANSACTION", the DDL will be failed
+        // so, just ignore the "START TRANSACTION" suffix
+        if (query.ends_with("START TRANSACTION"))
+        {
+            auto pos = query.rfind("START TRANSACTION");
+            if (pos > 0)
+                query.resize(pos);
+        }
     }
 
     void QueryEvent::dump(WriteBuffer & out) const

--- a/tests/integration/test_materialized_mysql_database/materialized_with_ddl.py
+++ b/tests/integration/test_materialized_mysql_database/materialized_with_ddl.py
@@ -2336,3 +2336,32 @@ def named_collections(clickhouse_node, mysql_node, service_name):
     )
     clickhouse_node.query(f"DROP DATABASE IF EXISTS {db}")
     mysql_node.query(f"DROP DATABASE IF EXISTS {db}")
+
+
+def create_table_as_select(clickhouse_node, mysql_node, service_name):
+    db = "create_table_as_select"
+    mysql_node.query(f"DROP DATABASE IF EXISTS {db}")
+    clickhouse_node.query(f"DROP DATABASE IF EXISTS {db}")
+    mysql_node.query(f"CREATE DATABASE {db}")
+    clickhouse_node.query(
+        f"CREATE DATABASE {db} ENGINE = MaterializeMySQL('{service_name}:3306', '{db}', 'root', 'clickhouse')"
+    )
+    mysql_node.query(
+        f"CREATE TABLE {db}.t1(a INT NOT NULL PRIMARY KEY) ENGINE = InnoDB"
+    )
+    mysql_node.query(f"INSERT INTO {db}.t1 VALUES (1)")
+    check_query(
+        clickhouse_node,
+        f"SHOW TABLES FROM {db} FORMAT TSV",
+        "t1\n",
+    )
+
+    mysql_node.query(f"CREATE TABLE {db}.t2(PRIMARY KEY(a)) AS SELECT * FROM {db}.t1")
+    check_query(
+        clickhouse_node,
+        f"SHOW TABLES FROM {db} FORMAT TSV",
+        "t1\nt2\n",
+    )
+
+    clickhouse_node.query(f"DROP DATABASE IF EXISTS {db}")
+    mysql_node.query(f"DROP DATABASE IF EXISTS {db}")

--- a/tests/integration/test_materialized_mysql_database/test.py
+++ b/tests/integration/test_materialized_mysql_database/test.py
@@ -529,3 +529,9 @@ def test_named_collections(started_cluster, started_mysql_8_0, clickhouse_node):
     materialized_with_ddl.named_collections(
         clickhouse_node, started_mysql_8_0, "mysql80"
     )
+
+
+def test_create_table_as_select(started_cluster, started_mysql_8_0, clickhouse_node):
+    materialized_with_ddl.create_table_as_select(
+        clickhouse_node, started_mysql_8_0, "mysql80"
+    )

